crates/qasm2/src/lex.rs

```diff
@@ -0,0 +1,772 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2023
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+//! The lexing logic for OpenQASM 2, responsible for turning a sequence of bytes into a
+//! lexed [TokenStream] for consumption by the parsing machinery.  The general strategy here is
+//! quite simple; for all the symbol-like tokens, the lexer can use a very simple single-byte
+//! lookahead to determine what token it needs to emit.  For keywords and identifiers, we just read
+//! the identifier in completely, then produce the right token once we see the end of the
+//! identifier characters.
+//!
+//! We effectively use a custom lexing mode to handle the version information after the `OPENQASM`
+//! keyword; the spec technically says that any real number is valid, but in reality that leads to
+//! weirdness like `200.0e-2` being a valid version specifier.  We do things with a custom
+//! context-dependent match after seeing an `OPENQASM` token, to avoid clashes with the general
+//! real-number tokenisation.
+
+use hashbrown::HashMap;
+use pyo3::prelude::*;
```

Comment:
From what I can tell, this only is using PyO3's prelude to import `PyResult`. Consider making that import explicit to make more clear the boundary between PyO3 vs normal Rust code. For example, this isn't dealing with the GIL etc. Separation of concerns

Reply:
Sure, done in 203c7bb9e.